domain/storageprovisioning/state/state.go

```diff
@@ -288,3 +315,108 @@ func (st *State) checkApplicationExists(
 
 	return true, nil
 }
+
+// GetStorageIDsForUnit returns the storage IDs for the given unit UUID.
+//
+// The following errors may be returned:
+// - [applicationerrors.UnitNotFound] when no unit exists for the supplied unit UUID.
+// - [corestorage.InvalidStorageID] when the provided unit UUID is invalid.
+func (s *State) GetStorageIDsForUnit(ctx context.Context, unitUUID string) ([]corestorage.ID, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	input := storageAttachment{
+		UnitUUID: unitUUID,
+	}
+	stmt, err := s.Prepare(`
+SELECT &storageAttachment.*
+FROM   storage_attachment sa
+JOIN   storage_instance si ON sa.storage_instance_uuid = si.uuid
+WHERE  unit_uuid = $storageAttachment.unit_uuid`, input)
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	var attachments storageAttachments
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if exists, err := s.checkUnitExists(ctx, tx, unitUUID); err != nil {
+			return errors.Capture(err)
+		} else if !exists {
+			return errors.Errorf("unit %q does not exist", unitUUID).Add(
+				applicationerrors.UnitNotFound,
+			)
+		}
+
+		err := tx.Query(ctx, stmt, input).GetAll(&attachments)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("unit %q does not exist", unitUUID).Add(
```

Comment:
This isn't correct, if no rows exists it just means that the unit doesn't have any storage attachments which is a very valid state. We have confirmed the unit exists above.

Can we get a regression test for this please.